### PR TITLE
api cleanup in preperation for identity improvements

### PR DESCRIPTION
**Pull request by tomerd**
motivation: infrastructure towards improving package identity correctness

changes:
* rename PackageReference::path to PackageReference::location
* make PackageReference::kind explicitly set by the call-site with no default and add static builder functions 
* rename PackageContainerConstraint::identifier to PackageContainerConstraint::package to align to true meaning
* rename PackageContainer::identifier to PackageContainer::package to align to true meaning
* adjust tests

### CODE REVIEW

**Comment by tomerd** [comment]
replaces some of the work done in #3152. this is the less risky subset of the changes, so we can create a separate PR for the more risky ones

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/PackageGraph/DependencyResolutionNode.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ public enum DependencyResolutionNode {
         // Don’t create a version lock for anything but a product.
         guard specificProduct != nil else { return nil }
         return PackageContainerConstraint(
-            container: package,
+            package: package,
```
Comment (neonichu) [inline Sources/PackageGraph/DependencyResolutionNode.swift:98]: Can we call this `packageReference`?

File: Sources/PackageGraph/LocalPackageContainer.swift [hunk 2]
```diff
@@ -24,35 +24,35 @@ import TSCUtility
 /// should be used as-is. Infact, they might not even have a git repository.
 /// Examples: Root packages, local dependencies, edited packages.
 public final class LocalPackageContainer: PackageContainer {
-    public let identifier: PackageReference
+    public let package: PackageReference
```
Comment (neonichu) [inline Sources/PackageGraph/LocalPackageContainer.swift:27]: Can we call this `packageReference`?

**Review by neonichu — APPROVED** [review]

File: Sources/PackageGraph/DependencyResolutionNode.swift (hunk 1 above)
Comment (tomerd) [reply Sources/PackageGraph/DependencyResolutionNode.swift:98]: @neonichu we could, but in many other places in the code vase it is called "package". I actually have a commit with this change but its large if we want to be consistent. wdyt?

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

**Comment by tomerd** [comment]
```
/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/branch-main/swift-corelibs-foundation/Sources/Foundation/Data.swift:25:8: error: missing required module 'SwiftOverlayShims'
15:17:50 import Glibc
```

**Pull request merged**